### PR TITLE
manually add turn.js

### PR DESCRIPTION
**Pull request by maruilian11**
**git repo url:** https://github.com/blasten/turn.js
**Watch 288
  Star 4,474
 Fork 1,581**
@Amomo could u help me check this PR for #7119 ? thank you!
**v0.1 is from https://github.com/blasten/turn.js/commit/9db239783f233119a36ecbe99ece174543a3617c
v0.11 is from https://github.com/blasten/turn.js/commit/9a978a2526ca9429ca5db35509aa0ed91f42a7de
v3 is from https://github.com/blasten/turn.js/commit/8b0e27214708c9ae0b09e858e6996ec8b44737da (only modified turn.js, so i minified it myself.)
v4 needs a commercial license, so i didn't add it.
v3's license is a little different from every BSD license, so i only regard it as BSD.**

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR 
 * [x] The lib has notable popularity 
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry 
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ ] Has valid tags for each versions (for git auto-update) 
https://github.com/blasten/turn.js/issues/547
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by PeterDaveHello** [comment]
Any reference to version 2?

**Comment by maruilian11** [comment]
the author seems only mention r3 and 4th version on official website and github repo.
https://github.com/blasten/turn.js/blob/master/readme.md
http://turnjs.com/docs/Download

**Comment by Amomo** [comment]
@maruilian11 sorry for late. Its license really weird.
I think this PR is ok, could you rebase it on our latest master branch?
Thank you!

**Comment by maruilian11** [comment]
@Amomo updated~

**Comment by Amomo** [comment]
@PeterDaveHello I think this PR is ok, could you check again? thank you~

File: ajax/libs/turn.js/3/turn.js [hunk 1]
```diff
@@ -0,0 +1,1917 @@
+/**
+ * turn.js
+ * turnjs.com
+ * Release: 3
+ *
+ * Copyright (C) 2012 Emmanuel Garcia
+ * Released under MIT License
```
Comment (PeterDaveHello) [inline ajax/libs/turn.js/3/turn.js]: It says MIT license.

File: ajax/libs/turn.js/0.1/turn.js [hunk 2]
```diff
@@ -0,0 +1,1290 @@
+/**
+ * turn.js
+ * turnjs.com
+ *
+ * Copyright (C) 2012 Emmanuel Garcia
+ * MIT Licensed
```
Comment (PeterDaveHello) [inline ajax/libs/turn.js/0.1/turn.js]: It says MIT license.

File: ajax/libs/turn.js/0.11/turn.js [hunk 3]
```diff
@@ -0,0 +1,1336 @@
+/**
+ * turn.js
+ * turnjs.com
+ *
+ * Copyright (C) 2012 Emmanuel Garcia
+ * MIT Licensed
```
Comment (PeterDaveHello) [inline ajax/libs/turn.js/0.11/turn.js]: It says MIT license.

File: ajax/libs/turn.js/package.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    "type": "git",
+    "url": "https://github.com/blasten/turn.js.git"
+  },
+  "keywords": [
+    "flipbooks",
+    "HTML5",
+    "CSS3"
+  ],
+  "author": "Emmanuel Garcia",
+  "license": "BSD",
```
Comment (PeterDaveHello) [inline ajax/libs/turn.js/package.json]: Is BSD correct?

Comment (maruilian11) [inline ajax/libs/turn.js/package.json]: 0.1 and 0.11 is under MIT
but v3 is under BSD
https://github.com/blasten/turn.js?files=1
http://turnjs.com/docs/Turn.js_3th_release_license_agreement_for_noncommercial_projects

Comment (PeterDaveHello) [inline ajax/libs/turn.js/package.json]: The license in the file says MIT, you should confirm that first.

**Comment by maruilian11** [comment]
@PeterDaveHello sorry, i found the author update license in https://github.com/blasten/turn.js/commit/8db4c68a16aca1d82bd0eedd8ead5c187772e4b9#diff-b030a457ef94fcccb143cb4dd57ae14e.
i have updated v3 files.

**Comment by PeterDaveHello** [comment]
You still have problem with your commit log, is `v0.1, v0.11, v3` a sentence? If not, why do you use a single line for it?

**Comment by maruilian11** [comment]
@PeterDaveHello i corrected it.

**Comment by Amomo** [comment]
ok... I think no problems in files , licenses and commit message now. 🤔

@maruilian11 could you rebase this PR with latest master branch? Thank you very much!

**Comment by maruilian11** [comment]
@Amomo @Piicksarn could you help me check this PR again? thanks!

**Pull request merged**